src/components/overlay/overlay.tsx

```diff
@@ -55,7 +56,10 @@ class Overlay extends Component<any, any> {
       clearTimeout(this._timeoutId);
       this._timeoutId = null;
     }
-    this.props.removePlayerClass(style.overlayActive);
+    // Remove the overlay-active class only when there is a single child
+    if (this.overlayRef.current?.childElementCount === 1) {
```

Comment:
i don't like this condition because it's for a specific use case and because i'd rather take this information from somewhere else instead of checking the children, if possible.
why, when the overlay is unmounting, is something is still visible on the player ? how does that happen ?

Reply:
as i mentioned in the description, it happens when you open the cvaa while the video is playing and it reaches a quiz question cp. so under the cvaa you have the question overlay. then, when you close the cvaa overlay, the overlay component is unmounted and is removing the overlay-active class which causes the current issue.